File: backend/internal/handlers/dto/milestone.go
package dto

import "time"

type MilestoneResponse struct {
	ID          uint    `json:"id"`
	Code        string  `json:"code"`
	Name        string  `json:"name"`
	Description string  `json:"description"`
	Type        string  `json:"type"`
	Threshold   int     `json:"threshold"`
	IconURL     *string `json:"iconUrl,omitempty"`
}

type UserAchievementResponse struct {
	Milestone  MilestoneResponse `json:"milestone"`
	AchievedAt time.Time         `json:"achievedAt"`
}

File: frontend/src/types/milestone.ts
export interface Milestone {
  id: number;
  code: string;
  name: string;
  description: string;
  type: 'SENT' | 'RECEIVED' | 'TOTAL';
  threshold: number;
  iconUrl?: string;
}

export interface UserAchievement {
  milestone: Milestone;
  achievedAt: string; // ISO date string
}
